pgxn/neon/libpagestore.c

```diff
@@ -250,12 +275,12 @@ load_shard_map(shardno_t shard_no, char *connstr_p, shardno_t *num_shards_p)
 	do
 	{
 		begin_update_counter = pg_atomic_read_u64(&pagestore_shared->begin_update_counter);
-		end_update_counter = pg_atomic_read_u64(&pagestore_shared->end_update_counter);
-
+		pg_read_barrier();
 		num_shards = shard_map->num_shards;
 		if (connstr_p && shard_no < MAX_SHARDS)
 			strlcpy(connstr_p, shard_map->connstring[shard_no], MAX_PAGESERVER_CONNSTRING_SIZE);
-		pg_memory_barrier();
+		pg_read_barrier();
+		end_update_counter = pg_atomic_read_u64(&pagestore_shared->end_update_counter);
```

Comment:
I don't think these changes are necessary, the old code looks correct to me. The new code is not wrong either but I don't see the point.